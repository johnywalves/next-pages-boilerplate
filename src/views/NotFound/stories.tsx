import { Story, Meta } from '@storybook/react/types-6-0'

import Error from '.'

export default {
  title: 'Error',
  component: Error
} as Meta

const Template: Story = (args) => <Error {...args} />

export const Basic = Template.bind({})
