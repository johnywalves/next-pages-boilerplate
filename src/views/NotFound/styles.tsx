import Link from 'next/link'

import styled from 'styled-components'

export const TitleNotFound = styled.h1`
  color: var(--color-primary);
`

export const SubtitleNotFound = styled.p`
  font-size: 1.5rem;
  margin: var(--20px) 0;
`

export const HomeNotFound = styled(Link)`
  font-size: 1.5rem;
`
