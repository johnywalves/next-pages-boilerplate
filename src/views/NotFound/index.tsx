import Button from 'components/Button'

import { TitleError, SubtitleError, HomeError } from './styles'

const NotFound = () => (
  <>
    <TitleError>Página não encontrada!</TitleError>
    <SubtitleError>
      Volte para paǵina anterior ou clique no logo para ir para a página
      principal
    </SubtitleError>
    <HomeError href="/" passHref>
      <Button>Ir para página inicial</Button>
    </HomeError>
  </>
)

export default NotFound
