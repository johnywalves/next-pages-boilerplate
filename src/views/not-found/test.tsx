import '@testing-library/jest-dom'

import { render, screen } from '@testing-library/react'

import NotFound from '.'

describe('<NotFound />', () => {
  it('should render the heading', () => {
    const { container } = render(<NotFound />)

    expect(
      screen.getByRole('heading', { name: /página não encontrada!/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<NotFound />)

    expect(container.firstChild).toHaveStyleRule(
      'color',
      'var(--color-primary-pure)'
    )
  })
})
