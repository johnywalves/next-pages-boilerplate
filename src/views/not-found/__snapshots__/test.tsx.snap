// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render the heading 1`] = `
.c0 {
  color: var(--color-primary-pure);
  text-align: center;
}

<h1
  class="c0"
>
  Página não encontrada!
</h1>
`;
